website/docs: Add docs for oauth2 back-channel logout (#15845)
goauthentik/authentik: website/docs/add-secure-apps/providers/oauth2/index.mdx

For advanced use cases, an authentik admin can use regular expressions (regex) instead of a redirect URL. For example, if you want to list ten different applications, instead of listing them all individually, you can create an expression with wildcards. When using regex, be aware that authentik uses a dot as a separator in the URL, but in regex a dot means "one of any character", a wildcard. You should therefore escape the dot with `\.` to prevent its interpretation as a wildcard.

> ### OAuth2/OpenID Connect Back-Channel Logout

Reviewer: ```suggestion
### OAuth2/OpenID Connect back-channel logout
```

> A server-to-server notification mechanism that allows an Identity Provider to notify connected OAuth2/OpenID clients when a user's session has been terminated.

Reviewer: ```suggestion
Using back-channel logout (a server-to-server notification mechanism) allows an identity provider to notify connected OAuth2/OpenID clients whenever a user's session is terminated.
```

> For more information, see [OAuth2/OpenID Connect Back-Channel Logout](./backchannel-logout.mdx).

Reviewer: ```suggestion
For more information, see our [OAuth2/OpenID Connect back-channel logout](./backchannel-logout.mdx) documentation.
```
